scipy/ndimage/tests/test_interpolation.py

```diff
@@ -1061,267 +1214,289 @@ def test_shift_nearest_boundary(self, order, prefilter):
-                                      [0, 1, 1, 0],
-                                      [0, 0, 0, 0]], dtype=np.float64)
-        expected = np.array([[0, 0, 0],
-                             [0, 1, 0],
-                             [0, 1, 0],
-                             [0, 0, 0]], dtype=np.float64)
+            data[:, :, i] = np.asarray([[0, 0, 0, 0],
+                                        [0, 1, 1, 0],
+                                        [0, 0, 0, 0]], dtype=np.float64)
+        data = xp.asarray(data)
+        expected = xp.asarray([[0, 0, 0],
+                               [0, 1, 0],
+                               [0, 1, 0],
+                               [0, 0, 0]], dtype=xp.float64)
         out = ndimage.rotate(data, 90, order=order)
         for i in range(3):
             assert_array_almost_equal(out[:, :, i], expected)
 
     @pytest.mark.parametrize('order', range(0, 6))
-    def test_rotate07(self, order):
-        data = np.array([[[0, 0, 0, 0, 0],
-                          [0, 1, 1, 0, 0],
-                          [0, 0, 0, 0, 0]]] * 2, dtype=np.float64)
-        data = data.transpose()
-        expected = np.array([[[0, 0, 0],
-                              [0, 1, 0],
-                              [0, 1, 0],
-                              [0, 0, 0],
-                              [0, 0, 0]]] * 2, dtype=np.float64)
-        expected = expected.transpose([2, 1, 0])
+    def test_rotate07(self, order, xp):
+        data = xp.asarray([[[0, 0, 0, 0, 0],
+                            [0, 1, 1, 0, 0],
+                            [0, 0, 0, 0, 0]]] * 2, dtype=xp.float64)
+        permute_dims = array_namespace(data).permute_dims
+        data = permute_dims(data, (2, 1, 0))
+        expected = xp.asarray([[[0, 0, 0],
+                                [0, 1, 0],
+                                [0, 1, 0],
+                                [0, 0, 0],
+                                [0, 0, 0]]] * 2, dtype=xp.float64)
+        expected = permute_dims(expected, (2, 1, 0))
         out = ndimage.rotate(data, 90, axes=(0, 1), order=order)
         assert_array_almost_equal(out, expected)
 
     @pytest.mark.parametrize('order', range(0, 6))
-    def test_rotate08(self, order):
-        data = np.array([[[0, 0, 0, 0, 0],
-                          [0, 1, 1, 0, 0],
-                          [0, 0, 0, 0, 0]]] * 2, dtype=np.float64)
-        data = data.transpose()
-        expected = np.array([[[0, 0, 1, 0, 0],
-                              [0, 0, 1, 0, 0],
-                              [0, 0, 0, 0, 0]]] * 2, dtype=np.float64)
-        expected = expected.transpose()
+    def test_rotate08(self, order, xp):
+        data = xp.asarray([[[0, 0, 0, 0, 0],
+                            [0, 1, 1, 0, 0],
+                            [0, 0, 0, 0, 0]]] * 2, dtype=xp.float64)
+        permute_dims = array_namespace(data).permute_dims
+        data = permute_dims(data, (2, 1, 0))  # == np.transpose
```

Comment:
This pattern seems okay when there is only one function to be accessed 👍. `xp_test` is probably cleaner if multiple functions are needed, though.